### PR TITLE
Emit a bunch of deprecation warnings for Gradle 8.1

### PR DESCRIPTION
**Pull request by eskatos**
* Closes https://github.com/gradle/gradle/issues/21902
* Closes https://github.com/gradle/gradle/issues/21882
* Closes https://github.com/gradle/gradle/issues/21900
* Closes https://github.com/gradle/gradle/issues/21914

----

This PR also opportunistically cleans up some smoke tests by removing `@UnsupportedWithConfigurationCache` annotation when it is not applicable anymore.

### CODE REVIEW

**Review by hythloda — APPROVED** [review]

File: subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy [hunk 1]
```diff
@@ -432,6 +432,12 @@ class JavaToolchainBuildOperationsIntegrationTest extends AbstractIntegrationSpe
 
         when:
         if (isKotlin1dot6) {
+            executer.beforeExecute {
```
[the remark this answers is not in the document]
Comment (eskatos) [reply subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy]: We have been not doing this in the past. The associated stacktraces point to where the problem comes from. Having a message talk about a given plugin when the actual deprecation comes from another would be confusing.

If we want to make this better we should instead make attribution better and have the deprecation warning better locate the incriminated user code. This is out of scope for this PR.

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Comment by eskatos** [comment]
@bot-gradle test and merge

File: subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (donat) [reply subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy]: 👍

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Pull request merged**